### DISCUSSION TITLE
Builders and webpack config for communication-react and communication-calling

### DISCUSSION THREAD

**Discussion by AdityaPune** [Azure/communication-ui-library#4396] [Q&A]
Hey all, I've been trying to work with Azure Communication Services for my application. Initially, after installing some of the azure libraries (calling, react, chat) I was getting issues with loaders saying I didn't have the appropriate loaders to work with these files. I was importing some functions (CallClient, CallComposite, etc) from them.

1. I had to include the azure package for my babel-loader in webpack config and add "process/browser" inside my resolve fallback. This has made my application build time slow. Is this correct or should I be doing something else?
2. I followed this storybook example: https://azure.github.io/communication-ui-library/?path=/docs/quickstarts-pstn--page However, my call adapter stays as undefined and there's an error inside my console saying CallClient is not a constructor. I'm pretty sure I'm passing the correct userId and token along with other args. This is mainly using the communication-react library
3. When I try it without the ui library and with just communication-calling's CallClient, I get the same error, saying CallClient isn't a constructor. In this scenario I also get a warning saying CallClient was not found in azure/communication-calling

Can someone help me with this or point me in the right direction?

**Comment by mgamis-msft** [comment]
Hi @AdityaPune,
Thank you so much for using ACS. Your questions here are very helpful to us.

1. Does you mean add the communication-react and communication-calling packages? I haven't added it to webpack in that way. Does running `npm install @azure/communication-react` and `npm install @azure/communication-calling` not work?
2. The call adapter should not be undefined with a proper userId and token. What locator object did you assign for your `AzureCommunicationCallAdapterArgs`? Just to confirm, you never see the configuration page like so:
![image](https://github.com/Azure/communication-ui-library/assets/79475487/ff86a9dd-b5ac-4f24-a4b3-81e0833b13ef)
